hermes/flake.nix

```diff
@@ -0,0 +1,93 @@
+{
+  description = "Rust implementation of the Inter-Blockchain Communication (IBC) protocol.";
+
+  inputs = {
+    # Nix Inputs
+    nixpkgs.url = "github:nixos/nixpkgs/nixos-unstable";
+
+    # Rust Inputs
+    rust-overlay.url = "github:oxalica/rust-overlay";
+    crate2nix = {
+      url = "github:yusdacra/crate2nix/feat/builtinfetchgit";
+      flake = false;
+    };
+    flake-utils.url = "github:numtide/flake-utils";
+
+    # Cosmos Sources
+    ibc-rs-src = {
+      url = github:informalsystems/ibc-rs;
+      flake = false;
+    };
+
+  };
+
+  outputs =
+    { nixpkgs
+    , rust-overlay
+    , crate2nix
+    , flake-utils
+    , ibc-rs-src
+    , ...
+    }:
+    let utils = flake-utils.lib; in
+    utils.eachDefaultSystem (system:
+    let
+      overlays = [
+        rust-overlay.overlay
+        (final: _: {
+          # Because rust-overlay bundles multiple rust packages into one
+          # derivation, specify that mega-bundle here, so that crate2nix
+          # will use them automatically.
+          rustc = final.rust-bin.stable.latest.default;
+          cargo = final.rust-bin.stable.latest.default;
+        })
+      ];
+      pkgs = import nixpkgs { inherit system overlays; };
+      evalPkgs = import nixpkgs { system = "x86_64-linux"; inherit overlays; };
+    in
+    rec {
+      packages = utils.flattenTree
+        {
+          hermes =
+            let
+              name = "ibc-rs";
+              # The use of `evalPkgs` here is a major hack. This is to get around the IFD limitation in nix flakes
```

Comment:
I am getting a warning:

```
trace: warning: crate2nix: Passing `buildRustCrate` as argument to Cargo.nix is deprecated. If you don't customize `buildRustCrate`, replace `callPackage ./Cargo.nix {}` by `import ./Cargo.nix { inherit pkgs; }`, and if you need to customize `buildRustCrate`, use `buildRustCrateForPkgs` instead.
```

Is it related to this workaround?

Reply:
Ah yeah, that's because I need to use a fork of `crate2nix`. Right now crate2nix does not use `fetchGit` under the hood, so it only looks for the `master` branch. `ibc-rs` uses `main` I believe.

```nix
crate2nix = {
      url = "github:yusdacra/crate2nix/feat/builtinfetchgit";
      flake = false;
    };
``` 

Here is the relevant issue upstream https://github.com/kolloch/crate2nix/issues/205

Reply:
I see. I thought the input for flakes is fixed, so inside a flake the Nix code shouldn't able to do `fetchGit` directly? Is the argument `ibc-rs-src` in the output closure just a URL, or is it a derivation containing the fetched source code already?

Reply:
So, `crate2nix` isn't a flake. I know they keep track of git resource hashes using a json file.

Here is the relevant code: https://github.com/yusdacra/crate2nix/blob/master/tools.nix#L234-L241